helm/portieris/templates/webhooks.yaml

```diff
@@ -2,11 +2,51 @@ apiVersion: admissionregistration.k8s.io/v1beta1
 kind: MutatingWebhookConfiguration
 metadata:
   name: image-admission-config
+  annotations:
   {{ if .Values.UseCertManager }}
+    cert-manager.io/inject-ca-from: {{ .Release.Namespace }}/portieris-certs
+  {{ end }}
+  namespace: {{ .Release.Namespace }}
+  labels:
+    app: {{ template "portieris.name" . }}
+    chart: {{ template "portieris.chart" . }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+webhooks:
+  - name: trust.hooks.securityenforcement.admission.cloud.ibm.com
+    clientConfig:
+      service:
+        name: {{ template "portieris.name" . }}
+        namespace: {{ .Release.Namespace }}
+        path: "/admit"
+      {{ if not .Values.UseCertManager }}
+      caBundle: {{ .Files.Get "certs/ca.crt" | b64enc }}
+      {{ end }}
+    rules:
+      - operations: [ "CREATE", "UPDATE" ]
+        apiGroups: ["*"]
+        apiVersions: ["*"]
+        resources: ["pods", "deployments", "replicationcontrollers", "replicasets", "daemonsets", "statefulsets", "jobs", "cronjobs"]
+    failurePolicy: Fail
```

Comment:
From the write-up in the issue, shouldn't this be "Ignore" (and then changed to "Fail" in the helm hook variant below)?

Reply:
if you have the admission of portieris work only by webhook faiure then you have to wait for it to timeout, the manifest version of the webhook here is not invoked due to the namespace selector which matches no namespaces and is then updated in the helm hook to be active

Reply:
Oh I see - so this approach creates a webhook that's guaranteed not to fire (unless you manage to set the namespace label explicitly to `notExist` before starting) , and then upgrade it post-install. Makes sense to me.

Reply:
yes, it just feels a bit ugly